### PR TITLE
Enabled proposer sig and randao verifications for init sync

### PR DESCRIPTION
**Pull request by terencechain**
This PR enables proposer and randao signature verifications for initial sync. Verified flame graph and the additional verifications don't carry much of a penalties. (ie 1% more per verification)

With this we need a new method to compute state root without verifying proposer and randao signature. `computesStateRoot` is the implementation

### CODE REVIEW

File: beacon-chain/core/state/transition.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func CalculateStateRoot(
 	}
 
 	// Execute per block transition.
-	stateCopy, err = processBlockNoVerify(ctx, stateCopy, signed)
+	stateCopy, err = computesStateRoot(ctx, stateCopy, signed)
```
Comment (0xKiwi) [inline beacon-chain/core/state/transition.go]: Kinda weird name convention, can it be changed to `computeStateRoot`?

File: beacon-chain/core/state/transition.go [hunk 2]
```diff
@@ -616,3 +616,40 @@ func ProcessEpochPrecompute(ctx context.Context, state *pb.BeaconState) (*pb.Bea
 	}
 	return state, nil
 }
+
+// computesStateRoot computes the state root of the block without verifying proposer signature
+// and randao.
+func computesStateRoot(
```
Comment (rauljordan) [inline beacon-chain/core/state/transition.go]: Should this be named computeStateRootNoVerify?

File: beacon-chain/core/state/transition.go [hunk 3]
```diff
@@ -358,9 +358,9 @@ func ProcessBlock(
...
 }
 
-// processBlockNoVerify creates a new, modified beacon state by applying block operation
+// processBlockNoVerifyAttSigs creates a new, modified beacon state by applying block operation
 // transformations as defined in the Ethereum Serenity specification. It does not validate
-// block signature.
+// block attestation signatures.
 //
 //
 // WARNING: This method does not verify proposer signature. This is used for proposer to compute state root
```
Comment (0xKiwi) [inline beacon-chain/core/state/transition.go]: This warning is no longer needed right?

File: beacon-chain/core/state/transition.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/core/state/transition.go]: I mean it's `computeStateRoot`, by default `computeStateRoot` does not verify any signature, it's purpose is to pass the state root to proposer to sign

**Pull request merged**